src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1625,97 +1649,160 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
+                // Add the claims extracted from the access token.
+                foreach (var claim in ticket.Identity.Claims) {
+                    endpointContext.Claims.Add(claim);
+                }
 
-                return;
-            }
+                await Options.Provider.ValidationEndpoint(endpointContext);
 
-            var notification = new ValidationEndpointContext(Context, Options, request, ticket);
+                // Flow the changes made to the authentication ticket.
+                ticket = endpointContext.AuthenticationTicket;
 
-            // Add the claims extracted from the access token.
-            foreach (var claim in ticket.Identity.Claims) {
-                notification.Claims.Add(claim);
-            }
+                if (endpointContext.HandledResponse) {
+                    return;
+                }
 
-            await Options.Provider.ValidationEndpoint(notification);
+                if (ticket.Properties.ExpiresUtc.HasValue && ticket.Properties.ExpiresUtc < Options.SystemClock.UtcNow) {
+                    Options.Logger.WriteVerbose("expired token");
+                    payload.Add("active", false);
+                }
 
-            // Flow the changes made to the authentication ticket.
-            ticket = notification.AuthenticationTicket;
+                //TODO: do more testing for token validity.
 
-            if (notification.HandledResponse) {
-                return;
-            }
+                else {
+                    payload.Add(OpenIdConnectConstants.Claims.Active, true);
 
-            var payload = new JObject();
+                    var scopes = ticket.GetScopes();
+                    if (scopes.Any()) {
+                        payload.Add(OpenIdConnectConstants.Claims.Scope, string.Join(" ", scopes));
+                    }
+                    
+                    payload.Add(OpenIdConnectConstants.Claims.ClientId, request.ClientId);
 
-            payload.Add("audiences", JArray.FromObject(ticket.Properties.GetAudiences()));
-            payload.Add("expires_in", ticket.Properties.ExpiresUtc.Value);
-            
-            payload.Add("claims", JArray.FromObject(
-                from claim in notification.Claims
-                select new { type = claim.Type, value = claim.Value }
-            ));
+                    // TODO: Re-examine whether ticket.Identity will always be the resource owner who authorized the token
+                    // and whether Username should always be returned
+                    if (!string.IsNullOrWhiteSpace(ticket.Identity.Name)) {
+                        payload.Add(OpenIdConnectConstants.Claims.Username, ticket.Identity.Name);
+                    }
 
-            var context = new ValidationEndpointResponseContext(Context, Options, payload);
-            await Options.Provider.ValidationEndpointResponse(context);
+                    // TODO: Determine the token type for the given token when not an access token
```

Comment:
I don't think there's a particular token_type value we can use for identity/refresh tokens. You can remove this TODO :+1: